portal-web/docroot/html/taglib/ui/page_iterator/start.jsp

```diff
@@ -410,6 +412,74 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 	}
 </script>
 
+<script data-senna-track="temporary" type="text/javascript">
+
+	var dropdown = document.getElementById("<%= namespace %>");
+	var button = dropdown.querySelector('.dropdown-toggle');
+	var list = dropdown.querySelector('.dropdown-menu');
+	var options = list.querySelectorAll('.dropdown-item');
+	var selectedItemValue = button.dataset.attribute;
+
+	function onButtonKeyDown(event) {
+		if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'Enter' || event.key === 'Space') {
+			event.preventDefault();
+			list.classList.add('show');
+			button.setAttribute('aria-expanded', 'true');
+			var selectedOption = null;
+
+			Array.from(options).find(function(option) {
+				if (option.id == selectedItemValue) {
+					selectedOption = option;
+				}
+			});
+
+			if (selectedOption) {
+				selectedOption.focus();
+			}
+		}
+	}
+
+	button.addEventListener('keydown',onButtonKeyDown );
+
+	function handleKeyEvents(event) {
+		var currentIndex = Array.from(options).indexOf(document.activeElement);
+
+		if (event.key === 'ArrowDown') {
+			event.preventDefault();
+			if (currentIndex < options.length - 1) {
+				options[currentIndex + 1].focus();
+			}
+		} else if (event.key === 'ArrowUp') {
+			event.preventDefault();
+			if (currentIndex > 0) {
+				options[currentIndex - 1].focus();
+			}
+		}
```

Comment:
We also need to add a check for the `Esc` key and close the dropdown and move focus to the trigger for that case. If we were fully matching the interaction of the Picker, we should also close the menu and move to the next tabbable element when tab is pressed. I think that's too much to try and handle here though.

Reply:
So, I was able to add an `Esc` key event but I got a few unexpected behaviors. 

1. As soon I press `Esc`, the focus disappears and only when I press `Tab`, it goes to the next element. 
2. If I have SR on, I have to press `Esc` key 2 times to make it disappear (see image below) . 

Any insights about that?

![Screenshot_20231003_171249](https://github.com/liferay-peds/liferay-portal/assets/47724428/df5abc56-04ce-4801-8a86-f8fb3736d852)

Reply:
For escape, you need to also manually set the focus on the button.